### ISSUE TITLE
Found with fuzzer: Touverac the giant shopkeeper attacks you without knowing your location?

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#369]
```
Suddenly, the dungeon collapses.
Touverac the giant shopkeeper attacks you without knowing your location?
Generating more information you may report:

#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140046493636544) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140046493636544) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140046493636544, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007f5f1d3e8476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007f5f1d3ce7f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x000055baca96e2ca in NH_abort () at end.c:236
#6  0x000055baca975343 in panic (str=0x55bacb4f2240 "%s") at end.c:802
#7  0x000055bacaeeec62 in impossible (s=0x55bacb490b20 "%s attacks you without knowing your location?") at pline.c:685
#8  0x000055bacabd3344 in wildmiss (mtmp=0x6110003c4640, mattk=0x6060006e5acf) at mhitu.c:426
#9  0x000055bacabe615d in mattacku (mtmp=0x6110003c4640) at mhitu.c:1195
#10 0x000055bacad8e7b8 in dochug (mtmp=0x6110003c4640) at monmove.c:1020
#11 0x000055bacad7874d in dochugw (mtmp=0x6110003c4640) at monmove.c:163
#12 0x000055bacace0378 in movemon () at mon.c:1602
#13 0x000055baca66d22c in moveloop (resuming=0 '\000') at allmain.c:247
#14 0x000055bacb375b40 in main (argc=0, argv=0x7ffc63dc25b8) at ../sys/unix/unixmain.c:353

```

**Comment by elunna** [comment]
Info:

```
(rr) p u.uz
$1 = {dnum = 0 '\000', dlevel = 5 '\005'}

(rr) p u.umonster
$2 = 754 (PM_NECROMANCER)

(rr) p u.umonnum
$3 = 754

(rr) p urace.malenum
$4 = 533 (PM_VAMPIRIC)

(rr) p moves
$5 = 3189406

(rr) p u.uundetected
$9 = 0

(rr) p uleft
$6 = (struct obj *) 0x0

(rr) p uright
$7 = (struct obj *) 0x0

(rr) p uarmc
$8 = (struct obj *) 0x0

```

**Comment by elunna** [comment]
My guess for what happened:
A shopkeeper tried to attack us and wildly missed.

Judging from the toplines output, the shopkeeper was a giant and had a clobber attack. Then tried to attack us again at the old location (pre clobber).

Old location: (19, 9)
New location: (20, 8)
Both spots are regular room floor tiles.

I noticed that mtmp had mconf on. So I was able to reproduce that as soon as I realized the shopkeeper was confused.

I polymorphed into an Umber Hulk (for some reason potions of confusion did not work to confuse the shopkeeper), confused the shk, then was easily able to trigger the bug by getting clobbered back a spot and then the shk tried to attack their old perception of where I was.
